flink-java/src/main/java/org/apache/flink/api/java/typeutils/TypeExtractor.java

```diff
@@ -680,10 +680,20 @@ private static void validateInputType(Type t, TypeInformation<?> inType) {
 		}
 	}
 	
-	private static void validateInputType(Class<?> baseClass, Class<?> clazz, int inputParamPos, TypeInformation<?> inType) {
+	private static void validateInputType(Class<?> baseClass, Class<?> clazz, int inputParamPos, TypeInformation<?> inTypeInfo) {
 		ArrayList<Type> typeHierarchy = new ArrayList<Type>();
+
+		// try to get generic parameter
+		Type inType;
+		try {
+			inType = getParameterType(baseClass, typeHierarchy, clazz, inputParamPos);
+		}
+		catch (Exception e) {
+			return; // skip input validation e.g. for raw types
```

Comment:
I know its annoying to change pull requests for these minor changes.
If you want you can change it and push it to `master`.